lib/iris/fileformats/netcdf.py

```diff
@@ -581,29 +548,50 @@ def _get_cf_var_data(cf_var, filename):
     return as_lazy_data(proxy, chunks=chunks)
 
 
+class OrderedAddableList(list):
+    # Used purely in actions debugging, to accumulate a record of which actions
+    # were activated.
+    # It replaces a set, so as to record the ordering of operations, with
+    # possible repeats, and it also numbers the entries.
+    # Actions routines invoke the 'add' method, which thus effectively converts
+    # a set.add into a list.append.
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._n_add = 0
+
+    def add(self, msg):
+        self._n_add += 1
+        n_add = self._n_add
+        self.append(f"#{n_add:03d} : {msg}")
+
+
 def _load_cube(engine, cf, cf_var, filename):
     from iris.cube import Cube
 
     """Create the cube associated with the CF-netCDF data variable."""
     data = _get_cf_var_data(cf_var, filename)
     cube = Cube(data)
 
-    # Reset the pyke inference engine.
+    # Reset the actions engine.
     engine.reset()
 
-    # Initialise pyke engine rule processing hooks.
+    # Initialise engine rule processing hooks.
     engine.cf_var = cf_var
     engine.cube = cube
     engine.cube_parts = {}
     engine.requires = {}
-    engine.rule_triggered = set()
+    engine.rule_triggered = OrderedAddableList()  # set()
```

Comment:
This comment could be clearer, maybe "formerly set()".

Reply:
:+1: 
I think it is explained in the definition of OrderedAddableList, 
so can I just remove the comment ?  I think it is probably just an editing leftover (reminder to self).